### PR TITLE
fix "tail 800 lines: journald" flake

### LINKED ISSUE #14362: tail 800 lines journald: incomplete output
Two flakes in one week:
```
# podman-remote [options] logs --tail 800 <CID>
...
...makes it only to 400-something, or perhaps even just zero lines
? Failure [3.295 seconds]
         Podman logs
         /var/tmp/go/src/github.com[/containers/podman/test/e2e/logs_test.go:26](https://github.com/containers/podman/blob/e874fc105c0fcfa291e256071be8b48dc4f347f8/test/e2e/logs_test.go#L26)
           tail 800 lines: journald [It]
           /var/tmp/go/src/github.com[/containers/podman/test/e2e/logs_test.go:116](https://github.com/containers/podman/blob/e874fc105c0fcfa291e256071be8b48dc4f347f8/test/e2e/logs_test.go#L116)
         
           Expected [long string] to have length 800
```

### Podman logs [It] tail 800 lines: journald 
* fedora-35 : int remote fedora-35 root host [remote]
  * PR #14272
    * [05-19 09:16](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4855020456771584/html/int-remote-fedora-35-root-host.log.html#t--tail-800-lines--journald--1)
    * [05-19 09:16](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4855020456771584/html/int-remote-fedora-35-root-host.log.html#t--tail-800-lines--journald--2)
    * [05-19 09:16](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4855020456771584/html/int-remote-fedora-35-root-host.log.html#t--tail-800-lines--journald--3)
* fedora-36 : int remote fedora-36 root host [remote]
  * PR #13808
    * [05-25 09:10](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5626273706803200/html/int-remote-fedora-36-root-host.log.html#t--tail-800-lines--journald--1)
    * [05-25 09:10](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5626273706803200/html/int-remote-fedora-36-root-host.log.html#t--tail-800-lines--journald--2)
    * [05-25 09:10](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5626273706803200/html/int-remote-fedora-36-root-host.log.html#t--tail-800-lines--journald--3)

### PR DESCRIPTION
**Pull request by Luap99**
The test calls podman run -d followed by podman logs. There is no
guarantee the the container or conmon has written all its output.
Adding an extra podman wait should fix this.

Fixes #14362

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by vrothberg — APPROVED** [review]
LGTM, nice catch!

**Comment by rhatdan** [comment]
Could have just removed the -d from command above, but this works wel also.  I am surprised this one is not firing all the time.
/lgtm
/hold

**Comment by Luap99** [comment]
No I do not want to remove the `-d`. The command prints 1000 lines and they should not be in the test logs by default.

**Comment by Luap99** [comment]
> You could redirect the output to /dev/null. But anyways...

Then the test will no longer work, it must print the lines in the container and we cannot redirect the output from the podman command inside the testing suite.

**Comment by edsantiago** [comment]
>No I do not want to remove the -d. The command prints 1000 lines and they should not be in the test logs by default.

They are anyway, or at least 800 of them are (the output of `podman logs` is 800 lines). I'm OK with the `-d`, but I'd like to see this reasoning in a comment so future maintainers don't have to have this same conversation. And I would especially like to see the same `wait` added to the earlier 99 test.

File: test/e2e/logs_test.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ var _ = Describe("Podman logs", func() {
 		It("tail 99 lines: "+log, func() {
 			skipIfJournaldInContainer()
 
-			logc := podmanTest.Podman([]string{"run", "--log-driver", log, "-dt", ALPINE, "sh", "-c", "echo podman; echo podman; echo podman"})
+			logc := podmanTest.Podman([]string{"run", "--log-driver", log, ALPINE, "sh", "-c", "echo podman; echo podman; echo podman"})
 			logc.WaitWithDefaultTimeout()
 			Expect(logc).To(Exit(0))
 			cid := logc.OutputToString()
```
Comment (edsantiago) [inline test/e2e/logs_test.go:108]: Whoops. Looks like you need to `run -n foo` instead and `log foo`, or find some other way to get the CID.

**Pull request merged**